### ISSUE TITLE
Include filter get empty array.

### ISSUE THREAD

**Issue by colinshen** [strongloop/loopback#1813]
In angular
```
Fund.find({
            filter:{
              include:{
                relation:"values",
                scope:{
                  order:"date DESC",
                  limit:1
                }
              }
            }
          },function  (funds) {
            console.log(funds);
          });
```
only one of the funds will get the value. the values of all other funds are empty array.
then i tried in API Explorer, 
I used `{"include":{"relation":"values","scope":{"limit":1}}}` as the filter.
I got the same thing as in angular.

**Comment by loay** [comment]
Can you provide more info for this case please?
Any specific repo to be referred to?
Thanks.

**Comment by colinshen** [comment]
I will make a repo later. Did I use the correct json for include filter?

**Comment by loay** [comment]
have a look at this document concerning include filter. It should help:
https://docs.strongloop.com/display/public/LB/Include+filter

**Comment by richardpringle** [comment]
Hey @colinshen, just a tip: whenever you're posting in code in comments on GitHub, you should use \``` to wrap your code. It will make it easier for everyone to read! Check out the [GitHub docs](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) for more info.

**Comment by colinshen** [comment]
@richardpringle thanks. I will do it next time.

**Comment by colinshen** [comment]
@loay here is the repo. 
https://github.com/colinshen/loopback-sandbox
this is the filter i used in explore {"include":{"relation":"values","scope":{"limit":2}}}. You will see there is only the first data that has values.

**Comment by loay** [comment]
@colinshen  Having a look. Thanks.

**Comment by colinshen** [comment]
I hope I did not use it in a wrong way. So,please check and give some suggestions.

**Comment by loay** [comment]
@colinshen 
it is all in the scope limit as I see it. 

When I play with the values in the filter I do get different results. so let's remove the filter completely. 

```
{"include":{"relation":"values"}}
```

```
[
  {
    "name": "Fund1",
    "id": 1,
    "values": [
      {
        "name": "value0",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 1,
        "fundId": 1
      },
      {
        "name": "value1",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 2,
        "fundId": 1
      },
      {
        "name": "value2",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 3,
        "fundId": 1
      },
      {
        "name": "value3",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 4,
        "fundId": 1
      },
      {
        "name": "value4",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 5,
        "fundId": 1
      },
      {
        "name": "value5",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 6,
        "fundId": 1
      },
      {
        "name": "value6",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 7,
        "fundId": 1
      },
      {
        "name": "value7",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 8,
        "fundId": 1
      },
      {
        "name": "value8",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 9,
        "fundId": 1
      },
      {
        "name": "value9",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 10,
        "fundId": 1
      }
    ]
  },
  {
    "name": "Fund2",
    "id": 2,
    "values": [
      {
        "name": "value0",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 11,
        "fundId": 2
      },
      {
        "name": "value1",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 12,
        "fundId": 2
      },
      {
        "name": "value2",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 13,
        "fundId": 2
      },
      {
        "name": "value3",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 14,
        "fundId": 2
      },
      {
        "name": "value4",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 15,
        "fundId": 2
      },
      {
        "name": "value5",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 16,
        "fundId": 2
      },
      {
        "name": "value6",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 17,
        "fundId": 2
      },
      {
        "name": "value7",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 18,
        "fundId": 2
      },
      {
        "name": "value8",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 19,
        "fundId": 2
      },
      {
        "name": "value9",
        "createdAt": "2015-11-13T15:55:29.834Z",
        "id": 20,
        "fundId": 2
      }
    ]
  }
]
```

**Comment by colinshen** [comment]
@loay That's true all other filter work correctly without limit. but when i use limit. I got this. You can see in fund2, the values array is empty. Is this a bug, or it originally looks like this. I mean I want to get one value for each fund.
```
{"include":{"relation":"values","scope":{"limit":1}}}
```
```
[
  {
    "name": "Fund1",
    "id": 1,
    "values": [
      {
        "name": "value0",
        "createdAt": "2015-11-13T18:29:48.538Z",
        "id": 1,
        "fundId": 1
      }
    ]
  },
  {
    "name": "Fund2",
    "id": 2,
    "values": []
  }
]
```

**Comment by loay** [comment]
@colinshen 
I got what you meant now. 
As far as I know LIMIT and ORDER are not really supported inside INCLUDE. 
But There is a work around. 

for example, you can make your function call the relation like that: 

```
{"include":{"relation":"values","scope": {"where": {"name": "value0"}}}}
```
That will give the first value for each fund. 

```
[
  {
    "name": "Fund1",
    "id": 1,
    "values": [
      {
        "name": "value0",
        "createdAt": "2015-11-13T19:18:15.639Z",
        "id": 1,
        "fundId": 1
      }
    ]
  },
  {
    "name": "Fund2",
    "id": 2,
    "values": [
      {
        "name": "value0",
        "createdAt": "2015-11-13T19:18:15.639Z",
        "id": 11,
        "fundId": 2
      }
    ]
  }
]
```

You can play in the where condition accordingly to get the results that you need.

**Comment by loay** [comment]
@colinshen 
Did that work out for you? If not please let me know. Thanks.

**Comment by colinshen** [comment]
I just removed that part from my site because it is not very important now and I really want to get the newest data automatically. After Limit and order working, I will do it again. Are you guys working on those small things now? Anyway, thanks a lot!!!

**Comment by loay** [comment]
@colinshen 
so if you do the same code:
```
{"include":{"relation":"values","scope": {"order":"date DESC", "limit":1}}}
```
It will simply give the highest value regardless of the fund, which is not what you want .. you want the highest value or the highest 2 values for each fund. 

```
[
  {
    "name": "Fund1",
    "id": 1,
    "values": []
  },
  {
    "name": "Fund2",
    "id": 2,
    "values": [
      {
        "name": "value0",
        "createdAt": "2015-11-17T20:29:16.071Z",
        "id": 11,
        "fundId": 2
      }
    ]
  }
]
```

So what you can do is query using

GET /Funds/{id}

and simply iterate through the ids and it will give the requires result for each id. Let me know.

**Comment by colinshen** [comment]
@loay ok,I got it . Thank you

**Comment by loay** [comment]
closes#1813
